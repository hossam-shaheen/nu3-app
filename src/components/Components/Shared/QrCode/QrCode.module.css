.qr-code {
    width: 135px;
}