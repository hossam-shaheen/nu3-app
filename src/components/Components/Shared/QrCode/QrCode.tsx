import { QRCodeSVG } from "qrcode.react";
import { FunctionComponent } from "react";
import classes from "./QrCode.module.css";

export const QrCode: FunctionComponent<{ url: string }> = ({
  url,
}): JSX.Element => {
  return (
    <div className={classes["qr-code"]}>
      {url && <QRCodeSVG value={`${url}`} data-testid="QR-Code-SVG" />}
    </div>
  );
};
export default QrCode;
