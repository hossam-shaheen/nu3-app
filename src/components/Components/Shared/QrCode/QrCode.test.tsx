import React from "react";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import QrCode from "./QrCode";

const url = "https://www.nu3.de/products/nu3-bio-acerola-pulver";
test("QrCode pass valid url : Renders", () => {
  render(<QrCode url={url} />);
  const svgElement = screen.queryByTestId("QR-Code-SVG");
  expect(svgElement).toBeInTheDocument();
});

test("QrCode pass invalid url : Renders", () => {
  render(<QrCode url="" />);
  const svgElement = screen.queryByTestId("QR-Code-SVG");
  expect(svgElement).not.toBeInTheDocument();
});

test("QrCode pass valid url: Snapshot", () => {
  const component = renderer.create(<QrCode url={url} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("QrCode pass invalid url: Snapshot", () => {
  const component = renderer.create(<QrCode url="" />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
