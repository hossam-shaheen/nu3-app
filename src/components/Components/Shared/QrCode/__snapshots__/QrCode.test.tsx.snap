// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QrCode pass invalid url: Snapshot 1`] = `
<div
  className="qr-code"
>
  
</div>
`;

exports[`QrCode pass valid url: Snapshot 1`] = `
<div
  className="qr-code"
>
  <svg
    data-testid="QR-Code-SVG"
    height={128}
    shapeRendering="crispEdges"
    viewBox="0 0 29 29"
    width={128}
  >
    <path
      d="M0,0 h29v29H0z"
      fill="#FFFFFF"
    />
    <path
      d="M0 0h7v1H0zM12 0h1v1H12zM16 0h1v1H16zM18 0h3v1H18zM22,0 h7v1H22zM0 1h1v1H0zM6 1h1v1H6zM8 1h1v1H8zM10 1h1v1H10zM12 1h1v1H12zM15 1h2v1H15zM20 1h1v1H20zM22 1h1v1H22zM28,1 h1v1H28zM0 2h1v1H0zM2 2h3v1H2zM6 2h1v1H6zM11 2h2v1H11zM14 2h3v1H14zM18 2h3v1H18zM22 2h1v1H22zM24 2h3v1H24zM28,2 h1v1H28zM0 3h1v1H0zM2 3h3v1H2zM6 3h1v1H6zM8 3h3v1H8zM13 3h2v1H13zM17 3h1v1H17zM19 3h1v1H19zM22 3h1v1H22zM24 3h3v1H24zM28,3 h1v1H28zM0 4h1v1H0zM2 4h3v1H2zM6 4h1v1H6zM9 4h1v1H9zM11 4h1v1H11zM19 4h2v1H19zM22 4h1v1H22zM24 4h3v1H24zM28,4 h1v1H28zM0 5h1v1H0zM6 5h1v1H6zM8 5h2v1H8zM13 5h1v1H13zM15 5h2v1H15zM22 5h1v1H22zM28,5 h1v1H28zM0 6h7v1H0zM8 6h1v1H8zM10 6h1v1H10zM12 6h1v1H12zM14 6h1v1H14zM16 6h1v1H16zM18 6h1v1H18zM20 6h1v1H20zM22,6 h7v1H22zM9 7h2v1H9zM14 7h1v1H14zM16 7h1v1H16zM19 7h1v1H19zM0 8h5v1H0zM6 8h4v1H6zM12 8h3v1H12zM17 8h1v1H17zM20 8h2v1H20zM23 8h1v1H23zM25 8h1v1H25zM27 8h1v1H27zM0 9h2v1H0zM3 9h1v1H3zM5 9h1v1H5zM7 9h2v1H7zM10 9h2v1H10zM16 9h1v1H16zM18 9h7v1H18zM28,9 h1v1H28zM0 10h5v1H0zM6 10h1v1H6zM10 10h1v1H10zM12 10h1v1H12zM14 10h5v1H14zM22 10h1v1H22zM0 11h1v1H0zM2 11h2v1H2zM7 11h2v1H7zM11 11h1v1H11zM14 11h2v1H14zM21 11h3v1H21zM25 11h1v1H25zM27 11h1v1H27zM0 12h1v1H0zM2 12h1v1H2zM6 12h2v1H6zM9 12h2v1H9zM13 12h2v1H13zM17 12h1v1H17zM20 12h1v1H20zM25 12h2v1H25zM3 13h3v1H3zM8 13h2v1H8zM11 13h1v1H11zM14 13h1v1H14zM16 13h1v1H16zM19 13h2v1H19zM22 13h1v1H22zM24 13h1v1H24zM28,13 h1v1H28zM1 14h2v1H1zM4 14h1v1H4zM6 14h2v1H6zM10 14h1v1H10zM12 14h2v1H12zM15 14h1v1H15zM20 14h1v1H20zM23 14h1v1H23zM25 14h2v1H25zM0 15h1v1H0zM3 15h1v1H3zM18 15h3v1H18zM23 15h2v1H23zM27 15h1v1H27zM5 16h5v1H5zM11 16h5v1H11zM17 16h1v1H17zM19 16h1v1H19zM21 16h1v1H21zM23 16h1v1H23zM25 16h2v1H25zM0 17h1v1H0zM2 17h1v1H2zM4 17h2v1H4zM7 17h2v1H7zM10 17h2v1H10zM13 17h1v1H13zM16 17h1v1H16zM19 17h1v1H19zM22 17h3v1H22zM26 17h1v1H26zM28,17 h1v1H28zM0 18h1v1H0zM4 18h1v1H4zM6 18h3v1H6zM10 18h1v1H10zM13 18h6v1H13zM20 18h2v1H20zM23 18h1v1H23zM26 18h1v1H26zM0 19h1v1H0zM3 19h2v1H3zM7 19h3v1H7zM14 19h3v1H14zM20 19h1v1H20zM22 19h3v1H22zM27 19h1v1H27zM0 20h1v1H0zM3 20h1v1H3zM5 20h2v1H5zM8 20h1v1H8zM10 20h1v1H10zM13 20h2v1H13zM17 20h2v1H17zM20 20h5v1H20zM26,20 h3v1H26zM8 21h4v1H8zM13 21h1v1H13zM17 21h2v1H17zM20 21h1v1H20zM24,21 h5v1H24zM0 22h7v1H0zM8 22h1v1H8zM10 22h1v1H10zM12 22h1v1H12zM14 22h4v1H14zM19 22h2v1H19zM22 22h1v1H22zM24 22h3v1H24zM0 23h1v1H0zM6 23h1v1H6zM12 23h1v1H12zM19 23h2v1H19zM24 23h1v1H24zM27,23 h2v1H27zM0 24h1v1H0zM2 24h3v1H2zM6 24h1v1H6zM8 24h2v1H8zM13 24h5v1H13zM20 24h5v1H20zM26 24h2v1H26zM0 25h1v1H0zM2 25h3v1H2zM6 25h1v1H6zM8 25h2v1H8zM11 25h1v1H11zM17 25h2v1H17zM21 25h1v1H21zM23 25h1v1H23zM25 25h2v1H25zM28,25 h1v1H28zM0 26h1v1H0zM2 26h3v1H2zM6 26h1v1H6zM8 26h1v1H8zM15 26h3v1H15zM19 26h1v1H19zM21 26h7v1H21zM0 27h1v1H0zM6 27h1v1H6zM8 27h1v1H8zM10 27h1v1H10zM14 27h3v1H14zM18 27h1v1H18zM20 27h1v1H20zM22 27h2v1H22zM25 27h1v1H25zM27 27h1v1H27zM0 28h7v1H0zM8 28h6v1H8zM15 28h3v1H15zM19 28h1v1H19zM21 28h6v1H21z"
      fill="#000000"
    />
  </svg>
</div>
`;
